### PR TITLE
Add layout-x/y and enhance documentation

### LINKED ISSUE #648: Nested layout calls are resetting the x/y positioning of their root node
Similarly to #643, we should not set the x/y position on the root node. It will always be 0, 0 in the non-nested case and in the nested case it should remain where it is.

### PR DESCRIPTION
**Pull request by chrisprice**
Fixes #648

### CODE REVIEW

File: site/src/components/layout/layout.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ The following code performs the required layout:
 {{{example-code}}}
 ```
 
-The flexbox layout will be computed, resulting in translate transforms being applied to each element. Group elements do not have width or height properties, so this information is added via `layout-width` and `layout-height` properties. The layout component also provides a convenience accessor for these values, e.g. `selection.layout('width')`.
+Nodes are positioned and sized appropriately using a combination of `x`, `y`, `width`, `height` and `transform` attributes depending on the whether the node is an `svg`, `rect` or other. Additionally `layout-width`, `layout-height`, `layout-x` and `layout-y` attributes are added which can be accessed using e.g. `selection.layout('width')`.
```
Comment (ColinEberhardt) [inline site/src/components/layout/layout.md]: Currently it adds `x` and `y` properties without the `layout-` prefix, which looks like a bug.

File: site/src/components/layout/layout.md [hunk 2]
```diff
@@ -81,3 +81,17 @@ svg.append('g')
       'flexDirection': 'row'
   });
 ```
+
+When building high performance charts e.g. when targeting 60fps, layout can be a source of performance problems. So in order to get the best performance, it's worth understanding some of its internals.
+
+At a high level, when a node is laid-out -
```
Comment (ColinEberhardt) [inline site/src/components/layout/layout.md]: You probably need to make it clear that you are referring to node layout within the context of our SVG / flexbox implementation.

File: src/layout/layout.js [hunk 3]
```diff
@@ -80,6 +82,17 @@ function computeDimensions(node) {
     }
 }
 
+function computePosition(node) {
+    if (node.hasAttribute('layout-x') && node.hasAttribute('layout-y')) {
```
Comment (ColinEberhardt) [inline src/layout/layout.js]: Ah - ok, ignore the previous comment!

File: site/src/components/layout/layout.md [hunk 4]
```diff
@@ -81,3 +81,17 @@ svg.append('g')
...
+
+* if it has not previously been laid-out it, its inner dimensions are measured. These are then combined with its `layout-style` and those of its descendants. After running these through the `css-layout` algorithm, the results are then applied node-by-node to the subtree.
+
+* if it has previously been laid-out, its inner dimensions are not be measured. Instead the `layout-width`, `layout-height`, `layout-x` and `layout-y` attributes set on it by the previous layout will be used as the basis for the algorithm. These are combined with the `layout-style` attributes of the node and its descendants then applied to the subtree as above.
```
Comment (ColinEberhardt) [inline site/src/components/layout/layout.md]: How does this effect re-sizing? if layout is invoked, then the browser resized, the inner dimensions could potentially change, however, according to this rule, the inner dimensions will not be re-measured.

**Comment by ColinEberhardt** [comment]
Really good documentation, thanks @chrisprice - just one point that I think requires a little clarification, and one question on the behaviour.

**Comment by ColinEberhardt** [comment]
LGTM ::+1::

**Comment by chrisprice** [comment]
Just waiting for the build to be fixed then!

**Pull request merged**